spec/services/stores/shared_examples/store.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+shared_examples 'store' do |identifier_model_type, form_store_keys|
+  let(:identifier_model) { create(identifier_model_type) }
+  let(:store) { described_class.new(identifier_model) }
+  let(:redis) { double }
+  let(:invalid_key_error) { "#{described_class}::InvalidKeyError".constantize }
+
+  before do
+    allow(RedisClient).to receive(:current).and_return(redis)
+  end
+
+  describe '#clear_stash' do
+    subject do
+      store.clear_stash(form_store_key)
+    end
+
+    let(:form_store_key) { nil }
+
+    context 'when form_store_key is nil' do
+      let(:form_store_key) { nil }
+
+      it 'returns an error' do
+        expect { subject }.to raise_error(invalid_key_error)
+      end
+    end
+
+    form_store_keys.each do |store_key|
+      context "when form_store_key is #{store_key}" do
+        let(:form_store_key) { store_key }
+
+        let(:value) { nil }
+
+        if described_class == Stores::UserStore
+          # TECH DEBT: [Stores::UserStore] This is error prone
+          let(:redis_key) { "#{identifier_model.id}_#{form_store_key}" }
```

Comment:
it gives

`1_user`

Whereby 
- `1` is an id number and not related to either an `user` or a `course` or a `provider`
- `user` is a form key not as a identifier

Reply:
https://trello.com/c/DCiFvwcr/398-bug-user-store-identifier-id-is-not-unique